src/ontology/mondo-ingest.Makefile

```diff
@@ -13,6 +13,28 @@
 MAPPINGSDIR=../mappings
 SKIP_HUGE=false
 
+####################################
+### Package management #############
+####################################
+.PHONY: pip
+pip:
```

Comment:
To make things DRY I moved repetitive `python3 -m pip install --upgrade pip setuptools` out of each individual dependency goal, and made this a prerequisite to those goals.

Only disadvantage is that if you run `make dependencies`, it'll run this for each command dependency listed, but it shouldn't take up much time.

Reply:
This is fine. Note that an alternative pattern which I have only started to use recently is this (typed without checking for errors but you will get the picture). The reason for this is that it is never a good idea to have a .PHONY goal being a dependency, because it really defeats the purpose of `make`:

```
pip:
    python3 -m pip install --upgrade pip setuptools
 
sssom:
     pip install -U sssom

dependencies:
     $(MAKE) pip
     $(MAKE) sssom
````

Reply:
Sounds good. I changed the whole thing to:

```make
.PHONY:
pip:
	python3 -m pip install --upgrade pip setuptools

.PHONY: oaklib
oaklib:
	 python3 -m pip install --upgrade $@

.PHONY: sssom
sssom:
	python3 -m pip install --upgrade $@

.PHONY: semsql
semsql:
	python3 -m pip install --upgrade $@

.PHONY: dependencies
dependencies:
	$(MAKE) pip
	$(MAKE) oaklib
	$(MAKE) sssom
	$(MAKE) semsql
```

I kind of don't like the repetitiveness of `python3 -m pip install --upgrade $@` in multiple goals there, but I have tried things like creating functions in `make` before and it's just a whole realm of voodoo that I'm not good at / not worth the time for this.

Reply:
If you want to reduce the code you can refactor it like this:

```
.PHONY: pip-%
pip-%:
	 python3 -m pip install --upgrade $*

.PHONY: dependencies
dependencies:
	$(MAKE) pip-pip
	$(MAKE) pip-setuptools
	$(MAKE) pip-oaklib
	$(MAKE) pip-sssom
	$(MAKE) pip-semsql
```

Reply:
Ash, so that's one way of functionalizing things.

I don't know if Harshad still has need for depending on just one of these packages, but the only one I currently need to depend on is `oaklib`. But it's probably not a harm to upgrade them all if I just want my "sync subclass" PR to depend on this `dependencies` goal instead.